src/tree/values/FormattingElement.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Represents phrasing content (e.g. <strong>, <a>, <em>), except text, within an HTML-text.
+ */
+class FormattingElement {
+	/**
+	 * Represents phrasing content (e.g. <strong>, <a>, <em>), except text, within an HTML-text.
+	 *
+	 * @param {string} tag the HTML-tag of this element (e.g. "strong", "a", etc.)
+	 * @param {number} start the start position of the tag within the text.
+	 * @param {number} end the end position of the tag within the text.
+	 * @param {Object} [attributes] the attributes (as key-value pairs, e.g. "href='...'" => { href: '...' } ).
+	 */
+	constructor( tag, start, end, attributes ) {
+		this.tag = tag;
+		this.attributes = attributes;
+		this.start = start;
+		this.end = end;
+	}
+
+	/**
+	 * Stringifies the given HTML-attributes to a string of " key=value" pairs.
+	 *
+	 * @param {Object} attributes the attributes to serialize.
+	 * @returns {string} the stringified attributes.
+	 */
+	getAttributeString( attributes ) {
```

Comment:
This always returns a space in front, which works because it is only used in `toHTML`, but is not very expected if one would use this function separately (not sure why). If that should not happen, maybe add an underscore in front of the name to mark it as private?

Reply:
You're right! I made this into a private function.